lib/graphql/fragment_cache/cache_key_builder.rb

```diff
@@ -113,8 +113,13 @@ def initialize(object: nil, query:, path:, **options)
 
       def build
         Digest::SHA1.hexdigest("#{schema_cache_key}/#{query_cache_key}").then do |base_key|
-          next base_key unless object
-          "#{base_key}/#{object_key(object)}"
+          if @options[:object_cache_key]
```

Comment:
I guess we could do something like

```ruby
def build
  Digest::SHA1.hexdigest("#{schema_cache_key}/#{query_cache_key}").then do |base_key|
    [base_key, object_key(object)].compact.join("/")
  end
end

def object_key(obj)
  obj&._graphql_cache_key || @options[:object_cache_key]
end
```

Reply:
Putting the @options check in `object_key` means that the Rails `cache_key` would also require that check. I think your first idea as good putting it the origin of building the entire key, but I'm happy to change it!

Rails would have to do this too:

```diff
module GraphQL
     # Extends key builder to use .expand_cache_key in Rails
     class CacheKeyBuilder
       def object_key(obj)
+        return @options[:object_cache_key]
         return obj.graphql_cache_key if obj.respond_to?(:graphql_cache_key)
```

---

Also in that code, the object's cache key would override the override, so I think it would have to be swapped to check the options first. 

```ruby
def object_key(obj)
  @options[:object_cache_key] || obj&._graphql_cache_key 
end
```

---

I think the array form is fine (though it's marginally a little slower 18.19k/ips vs 15.65k/ips), but the extra object (Array) allocations in performance code is something I'd personally avoid. Happy to change it if you'd like.

Reply:
Now when I think about it again—it makes sense to avoid an extra array since this operation is gonna happen multiple times per request. Merging it i as is, thank you so much! 💪